ext/mbstring/libmbfl/filters/mbfilter_uuencode.c

```diff
@@ -149,3 +159,87 @@ int mbfl_filt_conv_uudec(int c, mbfl_convert_filter * filter)
 	}
 	return 0;
 }
+
+/* Using mbstring to decode UUEncoded text is already deprecated
+ * However, to facilitate the move to the new, faster internal conversion interface,
+ * We will temporarily implement it for UUEncode */
+
+static size_t mb_uuencode_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state)
+{
+	unsigned char *p = *in, *e = p + *in_len;
+	uint32_t *out = buf, *limit = buf + bufsize;
+
+	unsigned int _state = *state & 0xFF;
+	unsigned int size = *state >> 8;
+
+	while (p < e && out < limit) {
+		unsigned char c = *p++;
+
+		switch (_state) {
+		case uudec_state_ground:
+			if (c == 'b') {
+				unsigned char *needle = (unsigned char*)uuenc_begin_text+1, *haystack = p;
+				while (haystack < e && *haystack == *needle) {
+					haystack++;
+					needle++;
+					if (*needle == '\0') {
+						p = haystack;
+						_state = uudec_state_until_newline;
+						goto next_iteration;
+					}
+				}
+				/* We didn't find "begin " */
+			}
+			break;
+
+		case uudec_state_until_newline:
+			if (c == '\n') {
+				_state = uudec_state_size;
+			}
+			break;
+
+		case uudec_state_size:
+			size = UUDEC(c);
+			_state = uudec_state_a;
+			break;
+
+		case uudec_state_a:
+			if ((e - p) < 4) {
+				p = e;
+				break;
+			}
+
+			unsigned int a = UUDEC(c);
+			unsigned int b = UUDEC(*p++);
+			unsigned int c = UUDEC(*p++);
+			unsigned int d = UUDEC(*p++);
+
+			if (size > 0) {
+				*out++ = ((a << 2) | (b >> 4)) & 0xFF;
+				size--;
+			}
+			if (size > 0) {
+				*out++ = ((b << 4) | (c >> 2)) & 0xFF;
+				size--;
+			}
+			if (size > 0) {
+				*out++ = ((c << 6) | d) & 0xFF;
+				size--;
+			}
+
+			_state = size ? uudec_state_a : uudec_state_skip_newline;
+			break;
+
+		case uudec_state_skip_newline:
```

Comment:
Yep, it does. Imitating the (goofy) behavior of the legacy code here.

Please note, the objective here is to provide a temporary implementation of UUEncode which matches the behavior of the old one (even bug-for-bug) so we can move ahead with using the new text conversion interface in other places.

If UUEncode was going to stay in mbstring permanently, I would want to fix it up. Since we already know it's going to be removed in a future release, my interest right now is just maintaining BC until it's time to delete it.